runtime/lua/vim/_defaults.lua

```diff
@@ -94,10 +94,19 @@ do
     { silent = true, expr = true, desc = ':help v_@-default' }
   )
 
-  --- Map |gx| to call |vim.ui.open| on the identifier under the cursor
+  --- Map |gx| to call |vim.ui.open| on the <cfile> at cursor.
   do
     local function do_open(uri)
-      local _, err = vim.ui.open(uri)
+      local cmd, err = vim.ui.open(uri)
+      local rv = cmd and cmd:wait(1000) or nil
+      if cmd and rv then
+        if rv.signal and rv.signal ~= 0 then
+          err = ('vim.ui.open: command timeout: %s'):format(vim.inspect(cmd.cmd))
```

Comment:
Checking `signal` to detect "timeout" seems hacky. Should we expose a `timeout=true` flag (or some other "status") on the `wait()` result?  @lewis6991

Reply:
Is it even sound since the signal could have originated from somewhere else? We could add it as a field to `SystemCompleted`?

Reply:
Hence "hacky" :) Just noticed that code=124 on timeout? https://github.com/neovim/neovim/blob/1d73ecda7f9be51371096989e5f1b351caf55d83/runtime/lua/vim/_system.lua#L283

Any reason not to set it to -1 ?